lib/chat_web/live/main_live/page/lobby.ex

```diff
@@ -91,22 +89,40 @@ defmodule ChatWeb.MainLive.Page.Lobby do
       PubSub.broadcast!(
         Chat.PubSub,
         @topic,
-        :room_request
+        {:room_request, room_hash, me |> Identity.pub_key() |> Utils.hash()}
       )
     end)
 
     socket
     |> assign_room_list()
   end
 
-  def approve_requests(socket) do
+  def approve_room_request(%{assigns: %{room_map: room_map}} = socket, room_hash, user_hash) do
+    if room_map[room_hash] do
```

Comment:
@sergey-lukianov Let's imagine the case where we have 10 online users in the room. Does it mean that everyone has the incoming room_hash in the `room_map`?

Reply:
yes, they have